### PR TITLE
Remove unsecure default password

### PR DESCRIPTION
**Pull request by tleguern**
Providing default passwords in a role is a bad idea in my opinion, most of the time users won't read the README or defaults/main and the insecure value will be kept.

What do you think?

### CODE REVIEW

**Comment by OndrejHome** [comment]
Hello @tleguern ,

Thank you for opening PR with suggestion about removal of pre-defined password for pcsd authentication.
Providing a default simple static password is most probably not a good idea and if there is a better way on generating/providing default password that would be significantly more secure that would be nice.

By just removing default password and requiring user to provide some password I think that it wouldn't make cluster less insecure. I'm just sure that it will annoy the existing and new users. Additionally the role is currently generating the password in deterministic way (by providing static password salt) which from security point of view can be also viewed as a problem. It however enables the role to report if password change is needed or not.

https://github.com/OndrejHome/ansible.ha-cluster-pacemaker/blob/b5e6eda0c3604b44b050f0465eabe27fe29ca6cf/tasks/main.yml#L81-L84

But instead of password removal I have few other suggestions that I believe would be more appropriate to address security concern here:
(NOTE these are just ideas that I haven't tested if they are possible with Ansible)
- generate safe random passwords for each node in way that will make role idempotent and working also when adding/removing cluster nodes
- print some kind of warning when default password is not overridden in attempt to educate user that have not checked the README and understood the consequences (maybe just `fail` module with `ignore_errors:true` would make it visible enough)
- Make it more clear in README that one of the strongly recommended options to change is password. Maybe making the section "Security considerations/hardening" or similar
- Implement option that will require "strong security" (which will be by default disabled) - require defining password, and ideally all other default options in role to make sure that user has not forgotten something, disallow configuration of insecure fence_vmware_soap, restrict firewall access to only cluster nodes instead of broadly enabling high-availability firewalld service, etc.
 - ...

If you have an use case where you would be able to present the change as beneficial that would be fantastic.

**In short:** In current state I wouldn't consider this PR for inclusion due to created inconveniences without significant gains. If you pursue to implement better way of improving situation with default password I will be glad to check it out and provide you with feedback. List of suggestions on what I consider 'better way' is provided above.

Thank you for interesting topic to think about!

Ondrej (온드레이)

**Comment by tleguern** [comment]
Hello, thank you for the detailed answer. I am not sure how feasible the first item in the list is without caching the password as a fact and I do not have the necessary knowledge/access combo for the fourth but the other two are both doable and good ideas.

As a general rules all the roles I write have a `mandatory` filter applied to passwords and sometimes I even add a call to `assert` or `fail` to enforce the point.

**Comment by tleguern** [comment]
I updated the pull request: now only a non aborting error is printed when the default password is kept. I included a new section in the README but left it a bit terse as I don't know much about fences.

**Comment by OndrejHome** [comment]
Hi @tleguern ,
Thank you for update and patience with my reply.
The task and 'Security Considerations' part looks OK to me to be included.

The part that comments out password in `defaults/main.yaml` causes error with undefined variable which at first is OK, but later this will break when the user should be created.

If that can be removed and only change to `tasks/main.yaml` and addition of `Security consideration` in README is present I will go ahead merge this. I can make the change myself but wanted your opinion on why the `defaults/main.yaml` and value in README for password was changed - old changes?

NOTE: as there was merge of other PR recently this PR would be best rebased on current `master` branch to avoid conflicts (in README).

**Comment by tleguern** [comment]
Hello, I just pushed a new diff. The commented out variable in `defaults/main.yml` and wrong value in the README are errors on my part.

**Comment by OndrejHome** [comment]
Hello Tristan,  @tleguern ,

Looks good to me, merged. Thank you for your contribution!
